### ISSUE TITLE
All the responses should follow the best practice structure

### ISSUE THREAD

**Issue by agentmilindu** [iamchathu/nodejs-api-starter#2]
Usually, the JSON responses sent from RESTful services have a common structure to follow,

```
{
   "data" : { 
     "someprop" : "somevalue",
     ....
   },
   "error" : [ ] 
}
```

Or 

```
{
   "data" : {},
   "error" : [ 
     {
        "code" : "4000",
        "message" : "User not authorized"
     }
   ] 
}
```

Shall we follow the same?

**Comment by iamchathu** [comment]
Let's have errors like it but with ERROR status code in http status

Eg 

422 for Unprocessable Entity etc...

But let's keep 200 for data and send data directly.

What's you idea @agentmilindu

**Comment by agentmilindu** [comment]
yeah, the HTTP error code should follow the proper HTTP codes. The code in our error object can be extensible. For example, we can have several reasons to be sending HTTP 400 for an endpoint and we can have codes like "4000", "4001", "4002", etc for these several reasons. Then we can easily track, categorize, etc with these codes.
